flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -427,25 +386,25 @@ public Builder maxPlanningSnapshotCount(int newMaxPlanningSnapshotCount) {
       return this;
     }
 
-    public Builder fromProperties(Map<String, String> properties) {
-      Configuration config = new Configuration();
-      properties.forEach(config::setString);
-
-      return this.useSnapshotId(config.get(SNAPSHOT_ID))
-          .caseSensitive(config.get(CASE_SENSITIVE))
-          .asOfTimestamp(config.get(AS_OF_TIMESTAMP))
-          .startingStrategy(config.get(STARTING_STRATEGY))
-          .startSnapshotTimestamp(config.get(START_SNAPSHOT_TIMESTAMP))
-          .startSnapshotId(config.get(START_SNAPSHOT_ID))
-          .endSnapshotId(config.get(END_SNAPSHOT_ID))
-          .splitSize(config.get(SPLIT_SIZE))
-          .splitLookback(config.get(SPLIT_LOOKBACK))
-          .splitOpenFileCost(config.get(SPLIT_FILE_OPEN_COST))
-          .streaming(config.get(STREAMING))
-          .monitorInterval(config.get(MONITOR_INTERVAL))
-          .nameMapping(properties.get(DEFAULT_NAME_MAPPING))
-          .includeColumnStats(config.get(INCLUDE_COLUMN_STATS))
-          .maxPlanningSnapshotCount(config.get(MAX_PLANNING_SNAPSHOT_COUNT));
+    public Builder fromProperties(
```

Comment:
`fromProperties` method name is not accurate anymore. not sure what is the better name yet. 2nd parameter is probably better to called `readOptions` (than `properties`) to be consistent with the code and doc.

Reply:
>fromProperties method name is not accurate anymore. not sure what is the better name yet

how about `settleConfig` ?

Reply:
what about `resolveConfig`?